app/controllers/users_controller.rb

```diff
@@ -27,4 +56,24 @@ def user_params
     params.require(:user).permit(:name, :email, :password,
       :password_confirmation)
   end
+
+  def logged_in_user
+    unless logged_in?
+      store_location
+      flash[:danger] = "Please log in."
+      redirect_to login_url
+    end
+  end
+
+  def correct_user
+    redirect_to(root_url) unless current_user?(@user)
+  end
+
+  def admin_user
+    redirect_to(root_url) unless current_user.admin?
+  end
+
+  def load_user
+    @user = User.find(params[:id])
```

Comment:
Sự khác nhau khi dùng find và find_by?
Check @user có tồn tại hay không nữa e nhé

Reply:
find trả về cái đã tồn tại trong hệ thống, không thì sẽ crash. con find_by nếu không tồn tại nó sẽ trả về nil. em nghĩ hàm sử dùng find vì khi không tìm thấy nó sẽ trả về lỗi , còn find_by nó vẫn trả về nil chứ không phải lỗi .

Reply:
dùng find by

nhớ xử lý nil
`return if @user`
`flash[:warning] = "User not found"`
`redirect_to root_path`